dora/core/common/src/test/java/alluxio/UfsUrlTest.java

```diff
@@ -118,13 +136,342 @@ public void hashCodeTest() {
     assertEquals(ufsUrl2.hashCode(), ufsUrl3.hashCode());
   }
 
+  @Test
+  public void pathWithWhiteSpacesNoException() throws IllegalArgumentException {
+    String[] paths = new String[]{
+        "file:/// ",
+        "file:///  ",
+        "file:/// path",
+        "file:///path ",
+        "file:///pa th",
+        "file:/// pa th ",
+        "file:///pa/ th",
+        "file:///pa / th",
+        "file:// pa / th "
+    };
+    for (String path : paths) {
+      UfsUrl ufsUrl = UfsUrl.createInstance(path);
+      assertEquals(path, ufsUrl.toString());
+    }
+  }
+
+  /**
+   * Tests the {@link UfsUrl#createInstance(String)} constructor for URL
+   * without scheme or with invalid scheme.
+   */
+  // TODO(Yichuan): Bring it back while the improved ufsurl class is merged.
+  @Ignore("Bring it back in the future.")
+  @Test
+  public void pathWithoutSchemeException() {
+    String path1 =  "/ ";
+    Exception e1 = assertThrows(IllegalArgumentException.class, () -> {
+      UfsUrl ufsUrl = UfsUrl.createInstance(path1);
+    });
+    assertTrue(e1.getMessage().contains(String.format("empty scheme: %s", path1)));
+
+    String path2 = "file:  / / /path";
+    Exception e2 = assertThrows(IllegalArgumentException.class, () -> {
+      UfsUrl ufsUrl = UfsUrl.createInstance(path2);
+    });
+    assertTrue(e2.getMessage().contains(String.format("empty scheme: %s", path2)));
+
+    String path3 =  "abc/:";
+    Exception e3 = assertThrows(IllegalArgumentException.class, () -> {
+      UfsUrl ufsUrl = UfsUrl.createInstance(path3);
+    });
+    assertTrue(e3.getMessage().contains(String.format("empty scheme: %s", path3)));
+  }
+
+  /**
+   * Tests the {@link UfsUrl#UfsUrl(String, String, String)} constructor to build a URI
+   * from its different components.
+   */
+  @Test
+  public void constructFromComponents() {
+    String scheme = "xyz";
+    String authority = "127.0.0.1:90909";
+    String path = "/a/b/c.txt";
+
+    UfsUrl uri1 = new UfsUrl(scheme, "", path);
+    assertEquals(scheme + "://" + path, uri1.toString());
+
+    UfsUrl uri2 = new UfsUrl(scheme, authority, path);
+    assertEquals(scheme + "://" + authority + path, uri2.toString());
+  }
+
+  /**
+   * Tests the {@link UfsUrl#equals(Object)} method.
+   */
+  @Test
+  public void equals() {
```

Comment:
Like it doesn't let you write a `hashCode()` I figure this one also overrides `Object.equals()`?
```suggestion
  public void equality() {
```